node/src/chain/header_validation.rs

```diff
@@ -178,7 +178,7 @@ impl<'a, DB: database::DB> Validator<'a, DB> {
         let (_, _, voters) = verify_block_att(
             self.prev_header.prev_block_hash,
             prev_block_seed,
-            self.provisioners.prev(),
```

Comment:
Passing the current list of provisioners instead of the previous block provisioners is actually breaking the block validation

Reply:
Depending on what `provisioners` contain, either this is wrong or `verify_success_att` is.

We need both the current list and the previous one to fully validate a block header, since there are two attestations: the one proving the Quorum on the candidate (`Attestation`) and the one proving the Quorum on the previous block (aka the block `Certificate`).

This could be simplified if we remove (and I hope we will) `Attestation` from the Block structure. But as of now, we can't avoid having to carry both provisioner lists in the Validator.

I would be in favor of adding `prev_provisioners` to the Validator structure, instead of using `ContextProvisioners` but the logic must be kept as before